rados/read_op.go

```diff
@@ -0,0 +1,57 @@
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <errno.h>
+// #include <stdlib.h>
+// #include <rados/librados.h>
+//
+import "C"
+
+import (
+	"unsafe"
+)
+
+// ReadOp manages a set of discrete object read actions that will be performed
+// together atomically.
+type ReadOp struct {
+	operation
+	op C.rados_read_op_t
+}
+
+// CreateReadOp returns a newly constructed read operation.
+func CreateReadOp() *ReadOp {
+	return &ReadOp{
+		op: C.rados_create_read_op(),
+	}
+}
+
+// Release the resources associated with this read operation.
+func (r *ReadOp) Release() {
+	C.rados_release_read_op(r.op)
+	r.freeElements()
+}
+
+// Operate will perform the operation(s).
+func (r *ReadOp) Operate(ioctx *IOContext, oid string, flags OperationFlags) error {
+	if err := ioctx.validate(); err != nil {
+		return err
+	}
+
+	cOid := C.CString(oid)
+	defer C.free(unsafe.Pointer(cOid))
+
+	r.reset()
+	ret := C.rados_read_op_operate(r.op, ioctx.ioctx, cOid, C.int(flags))
+	return r.finish("read", ret)
+}
+
+func (r *ReadOp) operateCompat(ioctx *IOContext, oid string) error {
```

Comment:
this looks like a copy from somewhere else, can the function not be inherited easier?

Reply:
Looks like you did this for the opElement interface. That is a nice cleanup, maybe it makes sense to do it for some of the other interfaces too.

Reply:
Sorry I don't understand. The line github shows your comment attached do doesn't seem to match what you're talking about for me....
Could you please elaborate some more?

Reply:
... not sure either what this refers to, sorry.